docs/topic/billing/things.md

```diff
@@ -0,0 +1,610 @@
+| Azure | $0.045 per hour + $0.045 per GiB | [link](https://azure.microsoft.com/en-us/pricing/details/azure-nat-gateway/) |
+
+Data *ingress* costs go from zero to $0.045 per GiB. So a user could download
+a terabyte of data, and it costs about $46 instead of 0$. This can escalate
+real quick, because this cost is *invisible* to the user.
+
+The shared IP also causes issues with external services that throttle per IP.
+
+So in general **we should completely avoid using NATs**, unless forced to by
+external constraints - primarily, some organizations might force a 'no public
+IPs on nodes' policy. While this may be useful policy in cases where individual
+users are creating individual VMs to do random things, it is counterproductive
+in our set up - and only makes things expensive.
+
+## Other Persistent Disk Storage
+
+We use dedicated, always on block storage for the following parts of
+the infrastructure:
+
+1. The hub database
+
+   This stores the `sqlite3` database that the JupyterHub
+   process itself uses for storing user information, server information, etc.
+   Since some of our hubs rely on admins explicitly adding users to the
+   hub database via the hub admin, this disk is important to preserve.
+
+   It is usually 1GiB in size, since that is the smallest disk we can get
+   from many of these cloud providers. However, on *Azure*, we also store
+   the hub logs on this disk for now - so the disk size is much larger.
+
+   High performance is not a consideration here, so we usually try to use
+   the cheapest (hence slowest disk) possible.
+
+2. Prometheus disk
+
+   We use [prometheus](https://prometheus.io/) to store time series data,
+   both for operational (debugging) as well as reporting (usage metrics)
+   purposes. We retain 1 year of data, and that requires some disk space.
+   We try to not use the slowest disk type, but this doesn't require the
+   fastest one either.
+
+   Currently, most prometheus disks are bigger than they need to be. We
+   can slim them down with effort if needed.
+
+3. Grafana disk
+
+   Grafana also uses a sqlite database the same way JupyterHub does. While
+   most of the dashboards we use are [managed in an external repo](https://github.com/jupyterhub/grafana-dashboards),
+   the database still holds user information as well as manually set up
+   dashboards. This does not need to be fast, and we also try to have the
+   smallest possible disk here (1GiB)
+
+4. Optional per-user disk for dedicated databases
+
+   One of the features we offer is a database per user, set up as a sidecar.
+   Most databases do not want to use NFS as the backing disk for storing
+   their data, and need a dedicated block storage device. One is created
+   *for each user* (via [Kubernetes Dynamic Volume Provisioning](https://kubernetes.io/docs/concepts/storage/dynamic-provisioning/)),
+   and this can get quite expensive
+   (for reasons outlined in the "Home Directory Storage" section above).
```

Comment:
```suggestion
   (for reasons outlined in the ["Home Directory Storage" section](ADD LINK) above).
```

Making this a link would be nice

Reply:
I've crosslinked this (and a few other similar references in the document)